subprojects/internal-instrumentation-api/src/main/java/org/gradle/internal/instrumentation/api/metadata/InstrumentationMetadata.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.instrumentation.api.metadata;
+
+import org.objectweb.asm.tree.MethodNode;
+
+public interface InstrumentationMetadata {
+    boolean isInstanceOf(String type, String superType);
+    MethodNode getMethodNode();
+}
```

Comment:
🤔 These two methods don't seem to belong to the same abstraction level...

Reply:
Split these two